src/melee.cpp

```diff
@@ -1870,11 +1870,11 @@ std::vector<special_attack> player::mutation_attacks(Creature &t) const
             }
 
             ///\EFFECT_UNARMED increases chance of attacking with mutated body parts
-
             ///\EFFECT_DEX increases chance of attacking with mutated body parts
-            const int chance = std::max( 1, mut_atk.chance - get_dex() - unarmed );
+            const float chance = get_dex() + unarmed / mut_atk.chance;
             add_msg( m_debug, "%s proc chance: %d", pr.first.c_str(), chance );
-            if( chance != 1 && !one_in( chance ) ) {
+            // If the mutation attack can fail to proc, and it does fail to proc, bail out
+            if( get_dex() + unarmed < mut_atk.chance && !x_in_y( get_dex() + unarmed, mut_atk.chance ) ) {
```

Comment:
`x_in_y( x, less_than_x )` will always succeed, there is no need to special case that.